packages/core/src/service/plugin/index.js

```diff
@@ -186,18 +217,54 @@ export default class PluginService {
       layout || {}
 
     if (contentName) {
-      const plugin = this.findContentPlugin(contentName, contentVersion)
-      newState.content = {
-        plugin,
-        state: plugin.unserialize(contentState)
+      const found = this.findContentPlugin(contentName, contentVersion)
+      if (!found.pluginWrongVersion || semver.lt(found.pluginWrongVersion.version, contentVersion)) {
+        // Standard case
+        newState.content = {
+          plugin: found.plugin,
+          state: found.plugin.unserialize(contentState)
+        }
+      } else {
+        // Attempt to migrate
+        const migratedState = this.migratePluginState(contentState, found.pluginWrongVersion, contentVersion)
+        if (found.pluginWrongVersion && migratedState) {
+          newState.content = {
+            plugin: found.pluginWrongVersion,
+            state: found.pluginWrongVersion.unserialize(migratedState)
+          }
+        } else {
+          // Unable to migrate, fallback to missing plugin
+          newState.content = {
+            plugin: found.plugin,
+            state: found.plugin.unserialize(contentState)
+          }
+        }
       }
     }
 
     if (layoutName) {
-      const plugin = this.findLayoutPlugin(layoutName, layoutVersion)
-      newState.layout = {
-        plugin,
-        state: plugin.unserialize(layoutState)
+      const found = this.findLayoutPlugin(layoutName, layoutVersion)
```

Comment:
This code block and the previous are very similar. Is it possible to merge those two?

Reply:
I'll try